ecosystem/indexer-grpc/indexer-transaction-generator/src/script_transaction_generator.rs

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    config::{ScriptStep, ScriptTransactionGeneratorConfig},
+    managed_node::ManagedNode,
+};
+use anyhow::{Context, Result};
+use aptos::{
+    account::fund::FundWithFaucet,
+    common::types::{CliCommand, MovePackageDir, ScriptFunctionArguments, TransactionOptions},
+    governance::CompileScriptFunction,
+    move_tool::{CompileScript, RunScript},
+};
+use aptos_protos::{
+    indexer::v1::{raw_data_client::RawDataClient, GetTransactionsRequest},
+    transaction::v1::Transaction,
+};
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+};
+
+/// GRPC request metadata key for the token ID.
+const LOCAL_INDEXER_GRPC_URL: &str = "http://127.0.0.1:50051";
+const TRANSACTION_STREAM_TIMEOUT_IN_SECS: u64 = 60;
+const DEFAULT_FUND_AMOUNT_IN_OCTA: u64 = 100_000_000;
+
+impl ScriptTransactionGeneratorConfig {
+    fn validate(&self) -> Result<()> {
+        // Validate the script transactions.
+        // 1. No output file names are duplicated.
+        let mut output_files = std::collections::HashSet::new();
+        for script_transactions in &self.scripted_transactions {
+            for step in &script_transactions.steps {
+                if let Some(output_name) = &step.output_name {
+                    if !output_files.insert(output_name) {
+                        return Err(anyhow::anyhow!(
+                            "[Script Transaction Generator] Output file name {} is duplicated",
```

Comment:
nit: we can provide more context like this

```                        "[Script Transaction Generator] Duplicate output file name '{}' found at transaction {} step {}.",```

Reply:
updated. 
skipped the version; since the version may not be that useful considering it's localnet and we wipe it for each run..